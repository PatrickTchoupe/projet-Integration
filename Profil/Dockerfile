FROM node:9-slim
WORKDIR /server
COPY package.json Server/server
RUN npm install
COPY . Server/server
EXPOSE 3000
CMD ["npm","start"]