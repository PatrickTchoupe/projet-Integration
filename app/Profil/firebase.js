export default
  {
    firebaseConfig: {
      apiKey: "AIzaSyBDfJQ6qgI8P9SR8-mQ6EK-2Ht11xjVDlY",
      authDomain: "helper1348.firebaseapp.com",
      databaseURL: "https://helper1348.firebaseio.com",
      projectId: "helper1348",
      storageBucket: "helper1348.appspot.com",
      messagingSenderId: "295532527033",
      appId: "1:295532527033:web:9b52add0968820d4244a68",
      measurementId: "G-P3RX0697EB"
    }
  }
