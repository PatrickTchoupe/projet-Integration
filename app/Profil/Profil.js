/* eslint-disable no-alert */
/* eslint-disable jsx-quotes */
/* eslint-disable no-use-before-define */
/* eslint-disable max-len */
/* eslint-disable quotes */
import React, { useEffect, useState } from 'react';
import {
  StyleSheet,
  Button,
  View,
  Image,
  Platform,
} from "react-native";
import { TextInput, Button as Test } from 'react-native-paper';
import * as ImagePicker from 'expo-image-picker';
import * as firebase from 'firebase';

//import App from './firebase'

// eslint-disable-next-line require-jsdoc
export default function Profil() {
  const [nom, setNom] = useState(' ');
  const [prenom, setPrenom] = useState(' ');
  const [adresse, setAdresse] = useState(' ');
  const [code, setCode] = useState(' ');
  const [userId, setId] = useState(0);
  const [bool, setBool] = useState(false);
  const [btnDisplay, setBtn] = useState(true);
  const [image, setImage] = useState(null);
  const [isPicked, setIsPicked] = useState(false);

 // firebase.initializeApp(firebaseConfig);
 // firebase.analytics();

 

  useEffect(() => {
    let data = localStorage.getItem('id');
    fetch('http://localhost:3000/users/'+data)
      .then((response) => response.json())
      .then((json) => {
        console.log(json[0]);
        setId(json[0].Id);
        setNom(json[0].Nom);
        setPrenom(json[0].Prenom);
        setAdresse(json[0].Adresse);
        setCode(json[0].CodePostal);
        setImage(json[0].PhotoProfil);
      })
      .catch((error) => {
        console.error(error);
      });
  }, []);

  const handleChangeName = (event) => {
    setNom(event.target.value);
  };
  const handleChangeSurname = (event) => {
    setPrenom(event.target.value);
  };
  const handleChangeAddress = (event) => {
    setAdresse(event.target.value);
  };
  const handleChangePostal = (event) => {
    setCode(event.target.value);
  };

  const pickImage = async () => {
    if (Platform.OS !== 'web') {
      const { status } = await ImagePicker.requestCameraRollPermissionsAsync();
      if (status !== 'granted') {
        alert('Sorry, we need camera roll permissions to make this work!');
      }
    }

    const result = await ImagePicker.launchImageLibraryAsync({
      mediaTypes: ImagePicker.MediaTypeOptions.All,
      allowsEditing: true,
      aspect: [4, 3],
      quality: 1,
    });

    console.log(result);
    if (!result.cancelled) {
      setImage(result.uri);
      setIsPicked(true);
    }
  };

  const uploadImage = async (uri, imgName) => {
    const response = await fetch(uri);
    const blob = await response.blob();

    const ref = firebase.storage().ref().child(imgName);
    ref.put(blob);

    const test = await firebase.storage()
      .ref(`/${imgName}`)
      .getDownloadURL();

    return test;
  };

  const handleSubmit = async () => {
    const imageName = `profile${userId}`;

    const test1 = isPicked ? await uploadImage(image, imageName) : image;

    const requestOptions = {
      method: 'PUT',
      headers:
      {
        'Content-type': 'application/json; charset=UTF-8',
        'Access-Control-Allow-Origin': '*',
      },
      body: JSON.stringify({
        Nom: nom,
        Prenom: prenom,
        CodePostal: code,
        Adresse: adresse,
        userId,
        Photo: test1,
      }),
    };
    console.log(requestOptions.body);

    fetch("http://localhost:3000/updateData", requestOptions)
      .then(() => ("Data successfully updated"))
      .catch((error) => {
        console.log(error);
      });

    setBtn(true);
    setBool(false);
  };

  const handleBoutonDisplay = () => {
    setBtn(false);
    setBool(true);
  };
  const handleCancel = () => {
    fetch('http://localhost:3000/users/2')
      .then((response) => response.json())
      .then((json) => {
        console.log(json[0]);
        setId(json[0].Id);
        setNom(json[0].Nom);
        setPrenom(json[0].Prenom);
        setAdresse(json[0].Adresse);
        setCode(json[0].CodePostal);
        setImage(json[0].PhotoProfil);
      })
      .catch((error) => {
        console.error(error);
      });
    setBtn(true);
    setBool(false);
  };

  return (
    <View style={styles.container}>
      <View style={styles.top}>
        <Image
          source={{ uri: image || 'http://ssl.gstatic.com/accounts/ui/avatar_2x.png' }}
          style={{
            width: 75,
            height: 75,
            borderRadius: 200 / 2,
          }}
        />
        {bool && 
              <Test onPress={pickImage} icon="camera" />
        }
      </View>

      <View>
        <TextInput
          value={nom}
          label="Nom"
          name="Nom"
          mode='outlined'
          editable={bool}
          onChange={handleChangeName}
        />
        <TextInput
          value={prenom}
          editable={bool}
          mode='outlined'
          label="Prenom"
          name="Prenom"
          onChange={handleChangeSurname}
        />
        <TextInput
          value={adresse}
          editable={bool}
          label="Adresse"
          name="adresse"
          mode='outlined'
          onChange={handleChangeAddress}

        />
        <TextInput
          value={code}
          editable={bool}
          label="Code postal"
          name="code"
          mode='outlined'
          onChange={handleChangePostal}
        />
      </View>
      <View style={styles.bottom}>
        {
          btnDisplay
          && (
            <Button
              title="Modifier"
              onPress={handleBoutonDisplay}
              color='blue'
            />
          )
        }
        {
          !btnDisplay
          && (
            <View style={{ flex: 2, padding: '10' }}>
              <Button
                title="Enregistrer"
                onPress={handleSubmit}
                color='green'
              />
            </View>
          )
        }
        <View style={{ flex: 0.1 }} />
        {
          !btnDisplay && (
            <View style={{ flex: 2, padding: '10' }}>
              <Button
                title="Annuler"
                onPress={handleCancel}
                color="red"
              />
            </View>
          )
        }
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "space-between",
    alignItems: 'center',
    backgroundColor: "#fff",
    padding: 20,
  },
  top: {
    flex: 0.3,
    borderTopLeftRadius: 20,
    borderTopRightRadius: 20,
    alignItems: 'center',
  },
  inputContainer: {
    flex: 0.3,
  },
  textInput: {
    padding: 10,
    textAlign: 'center',
    fontWeight: 'bold',
    borderWidth: 1,
    marginBottom: 10,
  },
  bottom: {
    flex: 0.3,
    flexDirection: "column",
    justifyContent: "space-between",
  },
});
