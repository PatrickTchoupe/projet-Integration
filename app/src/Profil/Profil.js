/* eslint-disable no-alert */

import React, {useEffect, useState} from 'react';
import {StyleSheet, Button, View, Image, TextInput, ActivityIndicator, Text, KeyboardAvoidingView} from 'react-native';
import NetInfo from '@react-native-community/netinfo';
import {Button as Test} from 'react-native-paper';
import * as ImagePicker from 'expo-image-picker';
import * as firebase from 'firebase';
import AsyncStorage from '@react-native-community/async-storage';

//import App from './firebase'

// eslint-disable-next-line require-jsdoc
export default function Profil({navigation: {navigate}}) {
  const [nom, setNom] = useState(' ');
  const [prenom, setPrenom] = useState(' ');
  const [adresse, setAdresse] = useState(' ');
  const [code, setCode] = useState(' ');
  const [userId, setId] = useState(0);
  const [bool, setBool] = useState(false);
  const [btnDisplay, setBtn] = useState(true);
  const [image, setImage] = useState(null);
  const [isPicked, setIsPicked] = useState(false);
  const [isLoaded, setLoaded] = useState(false);
  const [isConnected, setConnection] = useState(true);

  // firebase.initializeApp(firebaseConfig);
  // firebase.analytics();

  useEffect(() => {
    const unsubscribe = NetInfo.addEventListener((state) => {
      handleConnectivityChange(state.isConnected);
    });
    const retrieveData = async () => {
      try {
        const value = await AsyncStorage.getItem('user');
        if (value !== null) {
          // We have data!!
          const result = JSON.parse(value);
          return result.Id;
        }
      } catch (error) {
        // Error retrieving data
      }
    };
    const init = async () => {
      const test = await retrieveData();
      fetch(`https://help-recover-api.herokuapp.com/users/${test}`)
        .then((response) => response.json())
        .then((json) => {
          setLoaded(true);
          setNom(json[0].Nom);
          setId(json[0].Id);
          setPrenom(json[0].Prenom);
          setAdresse(json[0].Adresse);
          setCode(json[0].CodePostal);
          setImage(json[0].PhotoProfil);
        })
        .catch((error) => {
          console.error(error);
        });
    };
    unsubscribe();
    retrieveData();
    init();
  }, []);

  const handleConnectivityChange = (connect) => {
    setConnection({connect});
  };

  const pickImage = async () => {
    const permissionResult = await ImagePicker.requestCameraRollPermissionsAsync();

    if (permissionResult.granted === false) {
      alert('Permission to access camera roll is required!');
      return;
    }
    const result = await ImagePicker.launchImageLibraryAsync({
      mediaTypes: ImagePicker.MediaTypeOptions.All,
      allowsEditing: true,
      aspect: [4, 3],
      quality: 1,
    });
    if (!result.cancelled) {
      setImage(result.uri);
      setIsPicked(true);
    }
  };

  const deleteImage = () => {
    setImage('http://ssl.gstatic.com/accounts/ui/avatar_2x.png');
  };

  const uploadImage = async (uri, imgName) => {
    const response = await fetch(uri);
    const blob = await response.blob();

    const ref = firebase.storage().ref().child(imgName);
    ref.put(blob);

    const urlImage = await firebase.storage().ref(`/${imgName}`).getDownloadURL();

    return urlImage;
  };

  const handleSubmit = async () => {
    const imageName = `profile${userId}`;

    const temp = isPicked ? await uploadImage(image, imageName) : image;

    const requestOptions = {
      method: 'PUT',
      headers: {
        'Content-type': 'application/json; charset=UTF-8',
        'Access-Control-Allow-Origin': '*',
      },
      body: JSON.stringify({
        Nom: nom,
        Prenom: prenom,
        CodePostal: code,
        Adresse: adresse,
        userId,
        Photo: temp,
      }),
    };
    fetch('https://help-recover-api.herokuapp.com/updateData', requestOptions)
      .then()
      .catch((error) => {
        console.log(error);
      });

    setBtn(true);
    setBool(false);
  };

  const handleBoutonDisplay = () => {
    setBtn(false);
    setBool(true);
  };

  const handleCancel = () => {
    fetch(`https://help-recover-api.herokuapp.com/users/${userId}`)
      .then((response) => response.json())
      .then((json) => {
        setLoaded(true);
        setNom(json[0].Nom);
        setPrenom(json[0].Prenom);
        setAdresse(json[0].Adresse);
        setCode(json[0].CodePostal);
        setImage(json[0].PhotoProfil);
      })
      .catch((error) => {
        console.error(error);
      });
    setBtn(true);
    setBool(false);
  };

  return !isLoaded ? (
    isConnected ? (
      <ActivityIndicator size="large" color="blue" />
    ) : (
      <View>
        <Text>No internet Connection !</Text>
      </View>
    )
  ) : (
    <KeyboardAvoidingView style={styles.container}>
      <View style={styles.top}>
        <Image
          source={{uri: image || 'http://ssl.gstatic.com/accounts/ui/avatar_2x.png'}}
          style={{
            width: 75,
            height: 75,
            borderRadius: 200 / 2,
          }}
        />
        {bool && (
          <View style={styles.photoButton}>
            <Test onPress={pickImage} icon="camera" />
            <Test onPress={deleteImage} icon="delete" />
          </View>
        )}
      </View>

      <View>
        <TextInput
          value={nom}
          style={styles.textInput}
          name="Nom"
          editable={bool}
          onChangeText={(text) => setNom(text)}
        />
        <TextInput
          value={prenom}
          style={styles.textInput}
          editable={bool}
          name="Prenom"
          onChangeText={(text) => setPrenom(text)}
        />
        <TextInput
          value={adresse}
          style={styles.textInput}
          editable={bool}
          name="adresse"
          onChangeText={(text) => setAdresse(text)}
        />
        <TextInput
          value={code}
          style={styles.textInput}
          editable={bool}
          name="code"
          onChangeText={(text) => setCode(text)}
        />
      </View>
      <View style={styles.bottom}>
        {btnDisplay ? (
          <View style={styles.bottomButton}>
            <View style={styles.bottomButton1}>
              <Button title="Modifier" onPress={handleBoutonDisplay} color="blue" />
            </View>
            <View>
              <Button title="voir mes notations" onPress={() => navigate('Notation2')} />
            </View>
          </View>
        ) : (
          <View style={styles.bottomButton}>
            <View style={styles.bottomButton1}>
              <Button title="Enregistrer" onPress={handleSubmit} color="green" />
            </View>
            <View>
              <Button title="Annuler" onPress={handleCancel} color="red" />
            </View>
          </View>
        )}
      </View>
    </KeyboardAvoidingView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#fff',
  },
  photoButton: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  textInput: {
    width: 200,
    alignItems: 'center',
    alignContent: 'center',
    marginBottom: 10,
    color: 'black',
    paddingLeft: 75,
    paddingTop: 5,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: 'black',
  },
  top: {
    padding: 10,
  },
  bottom: {
    padding: 10,
    margin: 15,
  },
  bottomButton: {
    margin: 20,
    padding: 15,
  },
  bottomButton1: {
    margin: 5,
  },
});
