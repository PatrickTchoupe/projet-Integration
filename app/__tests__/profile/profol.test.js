/* eslint-disable import/imports-first */
/* eslint-disable import/no-unresolved */
/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import 'react-native';
import React from 'react';
import Profil from './../../Profil/Profil';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(<Profil />).toJSON();
    expect(tree).toMatchSnapshot();

});
