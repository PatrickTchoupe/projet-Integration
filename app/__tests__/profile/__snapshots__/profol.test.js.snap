// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-between",
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "flex": 0.3,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://ssl.gstatic.com/accounts/ui/avatar_2x.png",
        }
      }
      style={
        Object {
          "borderRadius": 100,
          "height": 75,
          "width": 75,
        }
      }
    />
  </View>
  <View>
    <View
      style={Object {}}
    >
      <View>
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "minHeight": 64,
              "paddingBottom": 0,
              "paddingTop": 8,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 3,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "bottom": 0,
                  "height": 2,
                  "left": 10,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                  "transform": Array [
                    Object {
                      "translateX": -3,
                    },
                  ],
                  "width": 8,
                }
              }
            />
            <Text
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "color": "transparent",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 18,
                  "opacity": 1,
                  "paddingHorizontal": 0,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 37,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": -30,
                    },
                    Object {
                      "scale": 0.75,
                    },
                    Object {
                      "scaleY": 0.2,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Nom
            </Text>
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#6200ee",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": -30,
                    },
                    Object {
                      "scale": 0.75,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Nom
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": -30,
                    },
                    Object {
                      "scale": 0.75,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Nom
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={false}
            multiline={false}
            name="Nom"
            onBlur={[Function]}
            onChange={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#6200ee"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "paddingHorizontal": 14,
                  "textAlign": "left",
                  "zIndex": 1,
                },
                Object {
                  "height": 56,
                },
                Object {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlignVertical": "center",
                },
                Array [
                  Object {},
                ],
              ]
            }
            underlineColorAndroid="transparent"
            value=" "
          />
        </View>
      </View>
    </View>
    <View
      style={Object {}}
    >
      <View>
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "minHeight": 64,
              "paddingBottom": 0,
              "paddingTop": 8,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 3,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "bottom": 0,
                  "height": 2,
                  "left": 10,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                  "transform": Array [
                    Object {
                      "translateX": -3,
                    },
                  ],
                  "width": 8,
                }
              }
            />
            <Text
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "color": "transparent",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 18,
                  "opacity": 1,
                  "paddingHorizontal": 0,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 37,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": -30,
                    },
                    Object {
                      "scale": 0.75,
                    },
                    Object {
                      "scaleY": 0.2,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Prenom
            </Text>
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#6200ee",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": -30,
                    },
                    Object {
                      "scale": 0.75,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Prenom
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": -30,
                    },
                    Object {
                      "scale": 0.75,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Prenom
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={false}
            multiline={false}
            name="Prenom"
            onBlur={[Function]}
            onChange={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#6200ee"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "paddingHorizontal": 14,
                  "textAlign": "left",
                  "zIndex": 1,
                },
                Object {
                  "height": 56,
                },
                Object {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlignVertical": "center",
                },
                Array [
                  Object {},
                ],
              ]
            }
            underlineColorAndroid="transparent"
            value=" "
          />
        </View>
      </View>
    </View>
    <View
      style={Object {}}
    >
      <View>
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "minHeight": 64,
              "paddingBottom": 0,
              "paddingTop": 8,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 3,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "bottom": 0,
                  "height": 2,
                  "left": 10,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                  "transform": Array [
                    Object {
                      "translateX": -3,
                    },
                  ],
                  "width": 8,
                }
              }
            />
            <Text
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "color": "transparent",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 18,
                  "opacity": 1,
                  "paddingHorizontal": 0,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 37,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": -30,
                    },
                    Object {
                      "scale": 0.75,
                    },
                    Object {
                      "scaleY": 0.2,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Adresse
            </Text>
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#6200ee",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": -30,
                    },
                    Object {
                      "scale": 0.75,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Adresse
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": -30,
                    },
                    Object {
                      "scale": 0.75,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Adresse
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={false}
            multiline={false}
            name="adresse"
            onBlur={[Function]}
            onChange={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#6200ee"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "paddingHorizontal": 14,
                  "textAlign": "left",
                  "zIndex": 1,
                },
                Object {
                  "height": 56,
                },
                Object {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlignVertical": "center",
                },
                Array [
                  Object {},
                ],
              ]
            }
            underlineColorAndroid="transparent"
            value=" "
          />
        </View>
      </View>
    </View>
    <View
      style={Object {}}
    >
      <View>
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "minHeight": 64,
              "paddingBottom": 0,
              "paddingTop": 8,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 3,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "bottom": 0,
                  "height": 2,
                  "left": 10,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                  "transform": Array [
                    Object {
                      "translateX": -3,
                    },
                  ],
                  "width": 8,
                }
              }
            />
            <Text
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "color": "transparent",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 18,
                  "opacity": 1,
                  "paddingHorizontal": 0,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 37,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": -30,
                    },
                    Object {
                      "scale": 0.75,
                    },
                    Object {
                      "scaleY": 0.2,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Code postal
            </Text>
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#6200ee",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": -30,
                    },
                    Object {
                      "scale": 0.75,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Code postal
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": -30,
                    },
                    Object {
                      "scale": 0.75,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Code postal
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={false}
            multiline={false}
            name="code"
            onBlur={[Function]}
            onChange={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#6200ee"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "paddingHorizontal": 14,
                  "textAlign": "left",
                  "zIndex": 1,
                },
                Object {
                  "height": 56,
                },
                Object {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlignVertical": "center",
                },
                Array [
                  Object {},
                ],
              ]
            }
            underlineColorAndroid="transparent"
            value=" "
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 0.3,
        "flexDirection": "column",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "blue",
              },
            ]
          }
        >
          Modifier
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 0.1,
        }
      }
    />
  </View>
</View>
`;
